### DISCUSSION TITLE
Failed to fetch API

### DISCUSSION THREAD

**Discussion by Luis-Paulo-Silva-Reis** [vercel/next.js#36203] [Help]
event - compiled successfully in 53 ms (74 modules)
[
  {
    message: 'Cannot query field "sourceUrl" on type "NodeWithFeaturedImageToMediaItemConnectionEdge".',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  },
  {
    message: 'Cannot query field "name" on type "NodeWithAuthorToUserConnectionEdge". Did you mean "node"?',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  },
  {
    message: 'Cannot query field "firstName" on type "NodeWithAuthorToUserConnectionEdge".',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  },
  {
    message: 'Cannot query field "lastName" on type "NodeWithAuthorToUserConnectionEdge".',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  },
  {
    message: 'Cannot query field "avatar" on type "NodeWithAuthorToUserConnectionEdge".',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  }
]
error - lib\api.js (24:10) @ fetchAPI
Error: Failed to fetch API
  22 |   if (json.errors) {
  23 |     console.error(json.errors)
> 24 |     throw new Error('Failed to fetch API')
     |          ^
  25 |   }
  26 |   return json.data
  27 | }

why this error ?

**Comment by mgiva** [comment]
Same issue for me to

[
  {
    message: 'Cannot query field "sourceUrl" on type "NodeWithFeaturedImageToMediaItemConnectionEdge".',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  },
  {
    message: 'Cannot query field "name" on type "NodeWithAuthorToUserConnectionEdge". Did you mean "node"?',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  },
  {
    message: 'Cannot query field "firstName" on type "NodeWithAuthorToUserConnectionEdge".',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  },
  {
    message: 'Cannot query field "lastName" on type "NodeWithAuthorToUserConnectionEdge".',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  },
  {
    message: 'Cannot query field "avatar" on type "NodeWithAuthorToUserConnectionEdge".',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  }
]

wait  - compiling /_error (client and server)...
event - compiled client and server successfully in 121 ms (468 modules)
error - lib\api.js (24:10) @ fetchAPI
Error: Failed to fetch API
  22 |   if (json.errors) {
  23 |     console.error(json.errors)
> 24 |     throw new Error('Failed to fetch API')
     |          ^
  25 |   }
  26 |   return json.data
  27 | }

**Comment by icyJoseph** [comment]
Hi,

How are you seeing this error? What are you trying to do? Do you have a reproduction repo link?

Are you perhaps trying to fetch data from `pages/api` from `getStaticProps` or `getServerSideProps`?

**Reply by icyJoseph** [reply]
@Luis-Paulo-Silva-Reis @mgiva - could you please double check and confirm, that you followed the WordPress steps as documented?

**Reply by Luis-Paulo-Silva-Reis** [reply]
Yes I follow I know the problem is from graphql

**Comment by dreallday** [comment]
the response from graphql in wordpress changed,

try adding the node around featuredImage, and author

```
 query AllPosts {
      posts(first: 20, where: {orderby: {field: DATE, order: DESC}}) {
        edges {
          node {
            title
            excerpt
            slug
            date
            featuredImage {
              node {
                sourceUrl
              }
            }
            author {
              node {
                name
                firstName
                lastName
                avatar {
                  url
                }
              }
            }
          }
        }
      }
    }
```


and around in the postBySlugId query

```
author {
        node {
          ...AuthorFields
        }
      }
```
